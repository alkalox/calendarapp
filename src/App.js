import React, { useState, useEffect} from 'react';
import {
  StyleSheet,
  ScrollView,
  Text,
} from 'react-native';

import CalendarButton from './components/CalendarButton'
import Event from './components/Event'
import Calendar from './components/Calendar'

import { markedDates, dataObj } from './utils/data'

getTodaysDate = () => {
  const date = new Date();
  let dateString = new Date(date.getTime() - (date.getTimezoneOffset() * 60000 ))
    .toISOString().split("T")[0];
  return dateString
}

const App = () => {
  const [showCalendar, setShowCalendar] = useState(false)
  // date defaults to today, else it is the date selected by the user
  const [date, setDate] = useState(getTodaysDate())
  const [dateHeading, setDateHeading] = useState(null)
  
  useEffect(() => {
    setFormattedDate(date)
  }, [date])

  const setFormattedDate = (date) => {
    let datesplit = date.split("-")
    setDateHeading(new Date(datesplit[0], datesplit[1] - 1, datesplit[2]).toDateString())
  }

  return (
    <ScrollView style = {styles.container}>
      <CalendarButton
        onPress = {() => setShowCalendar(!showCalendar)}
        showCalendar = {showCalendar} 
      />

      {showCalendar && 
        <Calendar
          date={date} 
          markedDates={markedDates}
          setDate={setDate}
        />
      }
      <Text style = {styles.headingStyle} testID = "dateHeading">
        {dateHeading}
      </Text>

      {dataObj[date] ? 
        dataObj[date].map(({time, name, title}, index) => (
          <Event time={time} name={name} title ={title} key={index} /> 
        )) 
      :
        <Text style = {styles.altTextStyle} testID = "noEventsText">
          ðŸ¤” There are no events for this date.
        </Text>
      }
    </ScrollView>
  );
};

const styles = StyleSheet.create({
  container: {
    backgroundColor:'#fff',
    flex: 1,
    padding: 20,
  },
  headingStyle: {
    fontSize: 22, 
    fontWeight: 'bold', 
    color: '#505050', 
    marginTop: 30
  },
  altTextStyle: {
    textAlign:'center', 
    marginTop: 50, 
    fontSize: 16
  }
})

export default App;
