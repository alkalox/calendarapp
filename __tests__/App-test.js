import 'react-native';
import React from 'react';
import App from '../src/App';
import { render, fireEvent } from '@testing-library/react-native'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('app renders correctly', () => {
  const tree = renderer.create(<App />).toJSON()
  expect(tree).toMatchSnapshot()
});

test('Calendar renders upon the calendar button being pressed', () => {
  const {debug, getByTestId, queryByTestId } = render(<App />)
  const calendarbutton = getByTestId('calendarbutton')
  // calendar component isn't rendered
  expect(queryByTestId('calendar')).toBeNull();
  fireEvent.press(calendarbutton)
  // assert that calendar component renders upon button being clicked
  expect(queryByTestId('calendar')).toBeTruthy();
})

test('Todays dateheading shows up by default', () => {
  const { getByTestId } = render(<App />)
  const dateheading = getByTestId('dateHeading').props.children
  const todaysDate = new Date().toDateString()
  expect(dateheading).toEqual(todaysDate)
})

test('Date is selected in Calendar and Events component shows up for it', () => {
  const { getByTestId, queryAllByTestId } = render(<App />)

  // press calendar btn to render calendar
  const calendarbutton = getByTestId('calendarbutton')
  fireEvent.press(calendarbutton)

  const testDate = '2020-11-15';
  
  // click on todays date button on the calendar
  const todaysDateBtn = getByTestId(`dayButton-${testDate}`)
  fireEvent.press(todaysDateBtn)

  // assert that events show up
  expect(queryAllByTestId('Events')).toBeTruthy()
  
})


