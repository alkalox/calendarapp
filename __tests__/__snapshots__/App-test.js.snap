// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app renders correctly 1`] = `
<RCTScrollView
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "padding": 20,
    }
  }
>
  <View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "opacity": 1,
        }
      }
      testID="calendarbutton"
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#000",
            "borderRadius": 20,
            "height": 70,
            "justifyContent": "center",
            "width": 70,
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../src/assets/calendar.png",
            }
          }
          style={
            Object {
              "height": 30,
              "width": 30,
            }
          }
        />
      </View>
      <Text
        style={
          Object {
            "color": "#000",
            "fontSize": 21,
            "fontWeight": "bold",
          }
        }
      >
        View full Calendar
      </Text>
      <Image
        source={
          Object {
            "testUri": "../../../src/assets/down-arrow.png",
          }
        }
        style={
          Object {
            "height": 23,
            "width": 23,
          }
        }
      />
    </View>
    <Text
      style={
        Object {
          "color": "#505050",
          "fontSize": 22,
          "fontWeight": "bold",
          "marginTop": 30,
        }
      }
      testID="dateHeading"
    />
    <View
      style={
        Object {
          "flexDirection": "row",
          "height": 90,
          "marginTop": 20,
          "width": "100%",
        }
      }
      testID="Events"
    >
      <View
        style={
          Object {
            "backgroundColor": "#518de7",
            "flex": 1,
          }
        }
      />
      <View
        style={
          Object {
            "flex": 30,
            "justifyContent": "space-between",
            "paddingLeft": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#74747b",
              "fontSize": 13,
            }
          }
        >
          8:00 AM - 8:30 AM
        </Text>
        <Text
          style={
            Object {
              "color": "#505050",
              "fontSize": 22,
              "fontWeight": "bold",
            }
          }
        >
          John
        </Text>
        <Text
          style={
            Object {
              "color": "#505050",
              "fontSize": 18,
            }
          }
        >
          Mindfulness Session
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#959595",
          "height": 1,
          "marginTop": 20,
        }
      }
    />
    <View
      style={
        Object {
          "flexDirection": "row",
          "height": 90,
          "marginTop": 20,
          "width": "100%",
        }
      }
      testID="Events"
    >
      <View
        style={
          Object {
            "backgroundColor": "#518de7",
            "flex": 1,
          }
        }
      />
      <View
        style={
          Object {
            "flex": 30,
            "justifyContent": "space-between",
            "paddingLeft": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#74747b",
              "fontSize": 13,
            }
          }
        >
          08:00 PM - 08:30 PM
        </Text>
        <Text
          style={
            Object {
              "color": "#505050",
              "fontSize": 22,
              "fontWeight": "bold",
            }
          }
        >
          Brad
        </Text>
        <Text
          style={
            Object {
              "color": "#505050",
              "fontSize": 18,
            }
          }
        >
          Reading Session
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#959595",
          "height": 1,
          "marginTop": 20,
        }
      }
    />
  </View>
</RCTScrollView>
`;
